src/libraries/Microsoft.Extensions.Options/src/ValidateOptionsResult.cs

```diff
@@ -39,27 +39,27 @@ public class ValidateOptionsResult
         /// <summary>
         /// Used to describe why validation failed.
         /// </summary>
-        public string FailureMessage { get; protected set; }
+        public string? FailureMessage { get; protected set; }
 
         /// <summary>
         /// Full list of failures (can be multiple).
         /// </summary>
-        public IEnumerable<string> Failures { get; protected set; }
+        public IEnumerable<string?> Failures { get; protected set; } = null!;
```

Comment:
Let's not do this.

```suggestion
        public IEnumerable<string?>? Failures { get; protected set; }
```

Reply:
How about?

```cs
[MemberNotNullWhen(true, nameof(Failures))]
public bool Failed { get; protected set; }
```

`Failures` are always set when fail result is generated (some code would NRE if it isn't)